src/agent/Core/ApplicationPool/Group/InitializationAndShutdown.cpp

```diff
@@ -130,9 +130,15 @@ Group::Group(Pool *_pool, const Options &_options)
 	}
 
 	detachedProcessesCheckerActive = false;
+	continueFlag = true;
+	boost::shared_ptr<bool> sharedContinueFlag( &continueFlag );
```

Comment:
I don't think this is correct. When sharedContinueFlag is destroyed, it will try to `delete &group->continueFlag` which will lead to memory corruption. If you just want the thread to check for cancellation, you can check `group->continueFlag` directly without using a shared\_ptr because the extra `shared_ptr<Group>` will keep both the group and the continueFlag alive.

But you shouldn't use a single boolean flag anyway because that's not thread-safe. Instead, you should use our system call interruption framework (see oxt/system_call.hpp). Inside timeoutRequestsCallback, call `oxt::sleep()` instead of `::sleep()`. When the Group is destroyed and calls `thread->interrupt_and_join()`, any sleeping `oxt::sleep()` calls will throw `boost::thread_interrupted`. Be sure to catch that exception. Just make sure that inside timeoutRequestsCallback you mark non-interruptable code parts with `boost::disable_interruption` and `boost::disable_syscall_interruption`.